examples/seq2seq/models.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2022 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""seq2seq example: Mode code."""
+
+# See issue #620.
+# pytype: disable=wrong-keyword-args
+
+import functools
+from typing import Any, Tuple
+
+from flax import linen as nn
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+PRNGKey = Any
+Shape = Tuple[int]
+Dtype = Any
+Array = Any
+
+class EncoderLSTM(nn.Module):
+  eos_id: int
+
+  @functools.partial(
+      nn.scan,
+      variable_broadcast='params',
+      in_axes=1,
+      out_axes=1,
+      split_rngs={'params': False})
+  @nn.compact
+  def __call__(self, carry, x):
+    lstm_state, is_eos = carry
+    new_lstm_state, y = nn.LSTMCell()(lstm_state, x)
+    # Pass forward the previous state if EOS has already been reached.
+    def select_carried_state(new_state, old_state):
+      return jnp.where(is_eos[:, np.newaxis], old_state, new_state)
+    # LSTM state is a tuple (c, h).
+    carried_lstm_state = tuple(
+        select_carried_state(*s) for s in zip(new_lstm_state, lstm_state))
+    # Update `is_eos`.
+    is_eos = jnp.logical_or(is_eos, x[:, self.eos_id])
+    return (carried_lstm_state, is_eos), y
+
+
+  @staticmethod
+  def initialize_carry(batch_size, hidden_size):
+    # use dummy key since default state init fn is just zeros.
+    return nn.LSTMCell.initialize_carry(
+        jax.random.PRNGKey(0), (batch_size,), hidden_size)
+
+
+class Encoder(nn.Module):
+  """LSTM encoder, returning state after EOS is input."""
```

Comment:
Can't parse this sentence; should `is` be `in` ?

Reply:
input is used as a verb, but I'll rephrase it to make it more clear.